### PR TITLE
Fix compatibility exclusions introduced by #11095 [KVL-1083]

### PR DESCRIPTION
**Pull request by fabiotudone-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by fabiotudone-da** [comment]
[Full compat test](https://dev.azure.com/digitalasset/daml/_build/results?buildId=90674&view=results) launched by @garyverhaegen-da on this PR

**Comment by garyverhaegen-da** [comment]
We'll get a clearer picture when the test is done running (including which versions are involved, which I find very hard to figure out from the streaming logs), but as a heads up we have a couple of

```
CommandDeduplicationIT
- [CommandDeduplicationIT:CDDeduplicateSubmitterBasic] Commands with identical submitter and command identifier should be deduplicated by the submission client ... Unexpected failure (ExpectedFailureException)
  Expected a failure when submitting a request as Alice for the second time, but got a successful result of: ()
```

in the logs already.

**Comment by garyverhaegen-da** [comment]
Found one combination for which we have that error: ledger-api-test-tool-1.16.0-platform-0.0.0.

**Comment by fabiotudone-da** [comment]
> This is neither one of those excluded nor changed (here or on #11095) though, which is puzzling.

Oh well, it actually is, as it has been fixed in #11095.

**Comment by garyverhaegen-da** [comment]
Linux build has finished now. It looks like this fails on old api-test-tools against HEAD platform:

```
//:ledger-api-test-tool-1.10.0-platform-0.0.0
//:ledger-api-test-tool-1.10.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.10.1-platform-0.0.0
//:ledger-api-test-tool-1.10.1-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.10.2-platform-0.0.0
//:ledger-api-test-tool-1.10.2-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.11.0-platform-0.0.0
//:ledger-api-test-tool-1.11.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.11.1-platform-0.0.0
//:ledger-api-test-tool-1.11.1-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.11.2-platform-0.0.0
//:ledger-api-test-tool-1.11.2-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.12.0-platform-0.0.0
//:ledger-api-test-tool-1.12.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.13.0-platform-0.0.0
//:ledger-api-test-tool-1.13.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.13.1-platform-0.0.0
//:ledger-api-test-tool-1.13.1-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.14.0-platform-0.0.0
//:ledger-api-test-tool-1.14.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.14.1-snapshot.20210706.7133.0.8ec16e94-platform-0.0.0
//:ledger-api-test-tool-1.14.1-snapshot.20210706.7133.0.8ec16e94-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.14.2-platform-0.0.0
//:ledger-api-test-tool-1.14.2-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.15.0-platform-0.0.0
//:ledger-api-test-tool-1.15.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.16.0-platform-0.0.0
//:ledger-api-test-tool-1.16.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.3.0-platform-0.0.0
//:ledger-api-test-tool-1.3.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.4.0-platform-0.0.0
//:ledger-api-test-tool-1.4.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.5.0-platform-0.0.0
//:ledger-api-test-tool-1.5.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.6.0-platform-0.0.0
//:ledger-api-test-tool-1.6.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.6.1-platform-0.0.0
//:ledger-api-test-tool-1.6.1-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.7.0-platform-0.0.0
//:ledger-api-test-tool-1.7.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.8.0-platform-0.0.0
//:ledger-api-test-tool-1.8.0-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.8.1-platform-0.0.0
//:ledger-api-test-tool-1.8.1-platform-0.0.0-postgresql
//:ledger-api-test-tool-1.9.0-platform-0.0.0
//:ledger-api-test-tool-1.9.0-platform-0.0.0-postgresql
```

**Comment by garyverhaegen-da** [comment]
test-tool 1.17.0 and above seem to work.

**Comment by fabiotudone-da** [comment]
@garyverhaegen-da 475c669 should actually introduce the correct exclusion. Let's see how this PR build goes before running a full compat test.

File: compatibility/bazel_tools/testing.bzl [hunk 1]
```diff
@@ -481,18 +483,42 @@ excluded_test_tool_tests = [
...
+            {
+                "start": "1.18.0-snapshot.20210928.7948.1",
+                "exclusions": [
                     "KVCommandDeduplicationIT:KVCommandDeduplicationCommitterDeduplication",
                 ],
             },
         ],
     },
+    {
+        "start": "1.13.0",
```
Comment (fabiotudone-da) [inline compatibility/bazel_tools/testing.bzl:511]:
```suggestion
        "start": "1.3.0",
```

Comment (fabiotudone-da) [reply compatibility/bazel_tools/testing.bzl:511]: I don't seem to get it right...

**Review by garyverhaegen-da — APPROVED** [review]
Manual compat test has succeeded, so I'll take the liberty of merging so we might have our first successful daily run of the week tomorrow.

**Pull request merged**